modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java

```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
+                String.format(SCREEN_QUERY_PARAM, LOGIN_SCREEN));
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(ORGANIZATION_TYPE))
+                .body("name", equalTo(tenant))
+                .body("screen", equalTo(LOGIN_SCREEN))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("add-custom-text.json")).
+                get("preference").toString());
+        JSONObject receivedPreference = new JSONObject(new JSONObject(response.asString()).
+                get("preference").toString());
+        Assert.assertTrue(areJSONObjectsEqual(expectedPreference, receivedPreference),
+                "The custom text preference schema of the Response is incorrect");
+    }
+
+    @Test(dependsOnMethods = {"testGetCustomTextPreference"})
+    public void testGetCustomTextPreferenceByQueryParams() throws IOException, JSONException {
+
+        Response response = getResponseOfGet(CUSTOM_TEXT_API_BASE_PATH + String.format
+                (CUSTOM_TEXT_COMPONENT_WITH_QUERY_PARAM, ORGANIZATION_TYPE, tenant, LOGIN_SCREEN, DEFAULT_LOCALE));
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(ORGANIZATION_TYPE))
+                .body("name", equalTo(tenant))
+                .body("screen", equalTo(LOGIN_SCREEN))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("add-custom-text.json")).
+                get("preference").toString());
+        JSONObject receivedPreference = new JSONObject(new JSONObject(response.asString()).
+                get("preference").toString());
+        Assert.assertTrue(areJSONObjectsEqual(expectedPreference, receivedPreference),
+                "The custom text preference schema of the Response is incorrect");
+    }
+
+    @Test(dependsOnMethods = {"testGetCustomTextPreferenceByQueryParams"})
+    public void testUpdateCustomTextPreference() throws IOException, JSONException {
+
+        String body = readResource("update-custom-text.json");
+        Response response = getResponseOfPut(CUSTOM_TEXT_API_BASE_PATH, body);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(ORGANIZATION_TYPE))
+                .body("name", equalTo(tenant))
+                .body("screen", equalTo(LOGIN_SCREEN))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("update-custom-text.json")).
```

Comment:
```suggestion
        JSONObject expectedPreference = new JSONObject(new JSONObject(body).
```

Reply:
Fixed with https://github.com/wso2/product-is/pull/16854/commits/f4920b1c7614e08eb8ca0aa4b15907e0c83be572